File: app/src/main/java/com/example/rodrigo/desafiomobile/gamesListDetail/GamesDetailActivity.java
package com.example.rodrigo.desafiomobile.gamesListDetail;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.rodrigo.desafiomobile.R;
import com.example.rodrigo.desafiomobile.entity.GamesEntity;
import com.squareup.picasso.Picasso;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.example.rodrigo.desafiomobile.gamesList.GamesListActivity.EXTRA;

public class GamesDetailActivity extends AppCompatActivity{

    @BindView(R.id.image_view_header)
    ImageView imageHeader;

    @BindView(R.id.text_view_nome)
    TextView nome;

    @BindView(R.id.gameData)
    TextView gameData;

    @BindView(R.id.gamePlataformas)
    TextView gamePlataformas;

    private GamesEntity gamesEntity;

    @Override
    protected void onCreate(Bundle savedInstaceState){
        super.onCreate(savedInstaceState);
        setContentView(R.layout.activity_games_detail);

        ButterKnife.bind(this);

        // Insere botão de voltar na ActionBar
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
        }

        Intent intent = getIntent();
        gamesEntity = (GamesEntity) intent.getSerializableExtra(EXTRA);

        Picasso.with(this)
                .load(gamesEntity.getImage())
                .centerCrop()
                .fit()
                .into(imageHeader);
        setTitle(gamesEntity.getName());
        nome.setText(gamesEntity.getName());
        gameData.setText(gamesEntity.getReleaseDate());

        String plataformas = textoPlataformas(gamesEntity.getPlatforms());
        gamePlataformas.setText(plataformas);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        switch (item.getItemId()){
            case android.R.id.home:
                onBackPressed();
                return true;

            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public String textoPlataformas(List<String> plataformas){
        String StringPlataformas = "";
        for(int i = 0; i < plataformas.size(); i++){
            if(i == 0)
                StringPlataformas = plataformas.get(i);
            else
                StringPlataformas = StringPlataformas + ", " + plataformas.get(i);
        }

        return StringPlataformas;
    }
}

File: app/src/main/java/com/example/rodrigo/desafiomobile/gamesListDetail/GamesDetailActivity.kt
package com.example.rodrigo.desafiomobile.gamesListDetail

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import android.widget.Toast
import com.example.rodrigo.desafiomobile.R
import com.example.rodrigo.desafiomobile.config.Config
import com.example.rodrigo.desafiomobile.entity.GamesEntity
import com.example.rodrigo.desafiomobile.gamesList.GamesListActivity
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubePlayer
import com.google.android.youtube.player.YouTubePlayerSupportFragment
import kotlinx.android.synthetic.main.activity_games_detail.*

class GamesDetailActivity : AppCompatActivity(), YouTubePlayer.OnInitializedListener {

    private lateinit var finalUrl: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_games_detail)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)

        val gamesEntity: GamesEntity = intent.getSerializableExtra(GamesListActivity.game) as GamesEntity

        title = gamesEntity.name
        text_view_name.text = gamesEntity.name
        gameDate.text = gamesEntity.releaseDate

        val platforms = textToPlatforms(gamesEntity.platforms)
        gamePlatforms.text = platforms

        val url = gamesEntity.trailer
        val urlSemEspaco = url.split("=")
        finalUrl = urlSemEspaco[urlSemEspaco.size - 1]

        val frag = supportFragmentManager.findFragmentById(R.id.youtube_fragment) as YouTubePlayerSupportFragment
        frag.initialize(Config.youTubeApiKey, this)
    }

    private fun textToPlatforms(platforms: List<String>): String{
        var stringPlatforms = StringBuilder()
        for (i in platforms.indices){
            if(i == 0)
                stringPlatforms = StringBuilder(platforms[i])
            else
                stringPlatforms.append(", ").append(platforms[i])

        }
        return stringPlatforms.toString()
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        return if (item?.itemId == android.R.id.home) {
            finish()
            true
        } else {
            super.onOptionsItemSelected(item)
        }
    }

    override fun onInitializationSuccess(p0: YouTubePlayer.Provider?, p1: YouTubePlayer?, p2: Boolean) {
        p1?.cueVideo(finalUrl)
    }
    override fun onInitializationFailure(p0: YouTubePlayer.Provider?, p1: YouTubeInitializationResult?) {
        Toast.makeText(this, "Erro ao reproduzir video", Toast.LENGTH_LONG).show()
    }

}